e2e/cp_test.go

```diff
@@ -1646,6 +1646,46 @@ func TestCopySingleS3ObjectIntoAnotherBucketWithPrefix(t *testing.T) {
 	assert.Assert(t, ensureS3Object(s3client, bucket, "prefix/"+filename, content))
 }
 
+// cp s3://bucket/* s3://dstbucket/
+func TestCopyAllObjectsIntoAnotherBucketIncludingSpecialCharacter(t *testing.T) {
+	t.Parallel()
+
+	const (
+		srcbucket = "bucket"
+		dstbucket = "dstbucket"
+	)
+
+	s3client, s5cmd, cleanup := setup(t)
+	defer cleanup()
+
+	createBucket(t, s3client, srcbucket)
+	createBucket(t, s3client, dstbucket)
+
+	filesToContent := map[string]string{
+		"sub&@$/test+1.txt":           "this is a test file 1",
+		"sub:,?/test; =2.txt":         "this is a test file 2",
+		"test&@$:,?;= 3.txt":          "this is a test file 3",
+		"sub/this-is-normal-file.txt": "this is a normal file",
+	}
+
+	for filename, content := range filesToContent {
+		putFile(t, s3client, srcbucket, filename, content)
+	}
+
+	src := fmt.Sprintf("s3://%v/*", srcbucket)
+	dst := fmt.Sprintf("s3://%v/", dstbucket)
+
+	cmd := s5cmd("cp", src, dst)
+	result := icmd.RunCmd(cmd)
+
```

Comment:
Could you please add an assertion for run command result ?
```suggestion
result.Assert(t, icmd.Success)
```

Reply:
I added the assertion